### ISSUE TITLE
 HiveError: This should not happen. Please open an issue on GitHub.

### ISSUE THREAD

**Issue by nayefradwi** [hivedb/hive#780]
**Steps to Reproduce**

I hot restarted after changing the type of the box from Map<String, dynamic> to dynamic and I got this issue. I tried to change back the type and hot restarting again nothing happened. I tried to stop and build again the issue is still there... 

**Code sample**
```
import 'dart:async';
import 'dart:developer';
import 'dart:isolate';

import 'package:driver_app/data_layer/models/responses.dart';
import 'package:driver_app/data_layer/models/rudderstack_event.dart';
import 'package:driver_app/data_layer/services/hive_db.dart';
import 'package:driver_app/data_layer/utils/Config.dart';
import 'package:driver_app/data_layer/utils/response.dart';
import 'package:hive/hive.dart';
import 'package:hive_flutter/hive_flutter.dart';
import 'package:path_provider/path_provider.dart';

import 'api/HttpClient.dart';

class RudderStackQueueWorker {
  static final RudderStackQueueWorker _queueWorker = RudderStackQueueWorker._();
  RudderStackQueueWorker._();
  Box<Map<String, dynamic>> eventsQueue;
  factory RudderStackQueueWorker.getInstance() => _queueWorker;
  bool isWorkerRunning = false;
  Isolate _workerIsolate;
  void startWorker() async {
    if (!isWorkerRunning) {
      eventsQueue = await Hive.openBox<Map<String, dynamic>>(
          HiveDbService.DRIVER_EVENTS_QUEUE_BOX);
      String auth = Config.rudderStackWriteKey + ":";
      String hivePath = (await getApplicationDocumentsDirectory()).path;
      _workerIsolate = await Isolate.spawn<List>(
        workerQueueHandler,
        [auth, "${Config.rudderStackDataPlaneUrl}/v1", hivePath],
      );
      isWorkerRunning = true;
    }
  }

  void trackEvent(RudderStackEvent event) {
    eventsQueue.add(event.toJson());
  }

  void stopWorker() {
    if (isWorkerRunning) {
      _workerIsolate.kill(priority: Isolate.immediate);
      isWorkerRunning = false;
      eventsQueue?.close();
    }
  }
}

void workerQueueHandler(List args) async {
  String auth = args[0];
  String baseRoute = args[1];
  Hive.init(args[2]);
  await HttpClient.getInstance().initializeRudderStack(auth, baseRoute);
  Timer.periodic(Duration(milliseconds: 250), (timer) async {
    // refresh the box
    Box<Map<String, dynamic>> eventsQueue;
    eventsQueue?.close();
    eventsQueue = await Hive.openBox<Map<String, dynamic>>(
        HiveDbService.DRIVER_EVENTS_QUEUE_BOX);

    // FIFO
    if (eventsQueue.length > 0) {
      var event = eventsQueue.getAt(0);
      await eventsQueue.deleteAt(0);
      handleMainThreadEvent(event).then((value) {
        if (value != null) eventsQueue.add(value.toJson());
      });
    }
  });
}

Future<RudderStackEvent> handleMainThreadEvent(var eventJson) async {
  RudderStackEvent event = RudderStackEvent.fromJson(eventJson);
  print(
    "trying event: ${event.properties['event']}, trial: ${event.retryCount}",
  );
  // try to track event
  InAppResponse response =
      await ResponseUtil.handleRudderStackResponse("/track", event.properties);
  if (response is InAppErrorResponse) {
    event.retryCount--;
    return event;
  }
  return null;
}

```

```
E/flutter (32452): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: HiveError: This should not happen. Please open an issue on GitHub.
E/flutter (32452): #0      BinaryReaderImpl.readFrame
package:hive/…/binary/binary_reader_impl.dart:250
E/flutter (32452): #1      FrameHelper.framesFromBytes
package:hive/…/binary/frame_helper.dart:21
E/flutter (32452): #2      FrameIoHelper.framesFromFile
package:hive/…/io/frame_io_helper.dart:42
E/flutter (32452): <asynchronous suspension>
E/flutter (32452): #3      StorageBackendVm.initialize
package:hive/…/vm/storage_backend_vm.dart:86
E/flutter (32452): <asynchronous suspension>
E/flutter (32452): #4      HiveImpl._openBox
package:hive/src/hive_impl.dart:116
E/flutter (32452): <asynchronous suspension>
E/flutter (32452): #5      HiveImpl.openBox
package:hive/src/hive_impl.dart:145
E/flutter (32452): <asynchronous suspension>
E/flutter (32452): #6      RudderStackQueueWorker.startWorker
package:driver_app/…/services/rudderstack_queue_worker.dart:25
E/flutter (32452): <asynchronous suspension>
E/flutter (32452):
```
**Version**
 - Platform: iOS, Android, Mac, Windows, Linux, Web
 - Flutter version: 2.2.3
 - Hive version: 2.0.4
 - hive_flutter: 1.1.0

**Comment by nayefradwi** [comment]
```
  HiveDbService._();
  static HiveDbService _hiveDbService = HiveDbService._();
  factory HiveDbService.getInstance() => _hiveDbService;

  // box keys
  static const String DRIVER_APP_DATES_BOX = "DriverAppDatesBox";
  static const String DRIVER_APP_STRINGS_BOX = "DriverAppStringsBox";
  static const String DRIVER_APP_FLAGS_BOX = "DriverAppFlagsBox";
  static const String DRIVER_APP_VALUES_BOX = "DriverAppValuesBox";
  static const String DRIVER_EVENTS_QUEUE_BOX = "EventsQueueBox";

  // value keys
  static const String ONLINE_PING_TIMESTAMP_KEY = "online_ping_timestamp";
  static const String WAS_LAST_ONLINE_KEY = "was_last_online";
  static const String LOGIN_SESSION_ID_KEY = "login_session_id";
  static const String ONLINE_SESSION_ID_KEY = "online_session_id";
  static const String BACKGROUND_ORDER_EXPIRY_DATE_KEY =
      "BACKGROUND_ORDER_EXPIRY_DATE_KEY";
  static const String BACKGROUND_ORDER_ID_KEY = "BACKGROUND_ORDER_ID_KEY";
  static const String IS_BACKGROUND_ORDER_A_TRIP = "IS_BACKGROUND_ORDER_A_TRIP";
  static const String TOTAL_DISTANCE = "TOTAL_DISTANCE";
  static const String TOTAL_TIME = "TOTAL_TIME";
  static const String POLYLINE = "POLYLINE";
  static const String TASK_DESCRIPTION = "TASK_DESCRIPTION";
  static const String PICKUP_LAT = "PICKUP_LAT";
  static const String PICKUP_LONG = "PICKUP_LONG";
  static const String DROPOFF_LAT = "DROPOFF_LAT";
  static const String DROPOFF_LONG = "DROPOFF_LONG";

  // default values
  static const NO_ONLINE_SESSION_YET = "NO_ONLINE_SESSION_YET";
  static const NO_LOGIN_SESSION_YET = "NO_LOGIN_SESSION_YET";

  Box<DateTime> _driverDatesBox;
  Box<String> _driverStringsBox;
  Box<bool> _driverFlagsBox;
  Box<num> _driverValuesBox;

  Future<void> intialize() async {
    try {
      await Hive.initFlutter();
      _driverDatesBox = await Hive.openBox<DateTime>(DRIVER_APP_DATES_BOX);
      _driverStringsBox = await Hive.openBox<String>(DRIVER_APP_STRINGS_BOX);
      _driverFlagsBox = await Hive.openBox<bool>(DRIVER_APP_FLAGS_BOX);
      _driverValuesBox = await Hive.openBox<num>(DRIVER_APP_VALUES_BOX);
    } catch (e) {
      RudderStackRouter.getInstance()
          .trackError(UnknownError(e.toString()), "intialize", "HiveDbService");
    }
  }
```

I am still getting the same issue and this code is run in the main ui isolate so its not on a separate isolate

**Comment by KurahaSho** [comment]
i have the same issue when init hive in workmanager, but in my case at the 1st excecution of the app works properly

**Comment by themisir** [comment]
Closed with #914.

Please upgrade to hive [v2.0.6](https://pub.dev/packages/hive/versions/2.0.6).
